tests/server/test_server_validation.py

```diff
@@ -21,6 +22,29 @@ def test_with_validator(both_fake_remote_clients):
     assert validator.valid
 
 
+def test_with_validator_json_response(both_fake_remote_clients, capsys):
+    """ Test that the validator writes compliant JSON when requested. """
+    from optimade.server.main_index import app
+
+    validator = ImplementationValidator(
+        client=both_fake_remote_clients,
+        index=both_fake_remote_clients.app == app,
+        respond_json=True,
+    )
+    try:
+        validator.validate_implementation()
+    except Exception:
+        print_exc()
+
+    captured = capsys.readouterr()
+    json_response = json.loads(captured.out)
+    assert json_response["failure_count"] == 0
+    assert json_response["internal_failure_count"] == 0
+    assert json_response["optional_failure_count"] == 0
+
```

Comment:
You could do some tests based on `validator` as well, specifically `validator.results`, I guess? Make sure the length of the `dict`s are the same, make sure they are the same as well actually, since I guess they should be?

Reply:
Could do, yup. There's not much else that can be compared consistently as success counts etc. will be changing every release

Reply:
Yeah, I guess this test is just to make sure you can retrieve that dictionary... :)